src/backend/app/Http/Controllers/Api/v1/ClientsController.php

```diff
@@ -3,32 +3,91 @@
 namespace App\Http\Controllers\Api\v1;
 
 use App\Http\Controllers\Controller;
-use Illuminate\Http\Request;
+use App\Http\Requests\DestroyClient;
+use App\Http\Requests\EditClient;
+use App\Http\Requests\IndexClient;
+use App\Http\Requests\StoreClient;
+use App\Http\Requests\UpdateClient;
+use App\Models\Client;
+use Illuminate\Http\Response;
+use Illuminate\Support\Collection;
 
 class ClientsController extends Controller
 {
-    //TODO: implement me pls
-    public function index(){}
+    /**
+     * Display data of all clients.
+     *
+     * @param IndexClient $request
+     * @return Collection
+     */
+    public function index(IndexClient $request) : Collection{
+        return Client::all();
+    }
 
     //TODO: implement me pls
-    public function create(){}
+    public function create() : Response{
+        return Response::create();
+    }
 
     //TODO: implement me pls
-    public function history(){}
+    public function history(IndexClient $request) : Response{
+        return Response::create();
+    }
 
-    //TODO: implement me pls
-    public function store(){}
+    /**
+     * Store a newly created client in storage.
+     *
+     * @param StoreClient $request
+     * @return JsonResponse
+     */
+    public function store(StoreClient $request) : JsonResponse{
+        $sanitized = $request->validated();
+        $client = Client::create($sanitized);
+        return response()->json($client, 201);
+    }
 
     //TODO: implement me pls
-    public function edit(){}
+    public function edit(EditClient $request, Client $client) : Response{
+        return Response::create();
+    }
 
-    //TODO: implement me pls
-    public function update(){}
+    /**
+     * Update the specified client in storage.
+     *
+     * @param UpdateClient $request
+     * @param int $clientId
+     * @return JsonResponse
+     */
+    public function update(UpdateClient $request, int $clientId) : JsonResponse{
+        $client = Client::findOrFail($clientId);
+        $data = $request->validated();
+        $client->update($data);
+        return response()->json($client, 200);
+    }
 
-    //TODO: implement me pls
-    public function destroy(){}
+    /**
+     * Remove the specified client from storage.
+     *
+     * @param DestroyClient $request
+     * @param int $clientId
+     * @return JsonResponse
+     */
+    public function destroy(DestroyClient $request, int $clientId) : JsonResponse{
+        $client = Client::findOrFail($clientId);
+        $client->delete();
+        return response()->json(null, 204);
+    }
 
-    //TODO: implement me pls
-    public function findClient(){}
+    /**
+     * Display the specified client.
+     *
+     * @param String $string
+     * @return JsonResponse
+     */
+    public function findClient(String $string) : JsonResponse{
+        return response()->json(Client::with('clients')->where('first_name', 'LIKE', '%' . $string . '%')
```

Comment:
tu odporucam pouzit ```ilike``` lebo ten je case insensitive

Reply:
a zaroven asi nechapem preco si pouzil ```Client::with('clients')``` ? tam taky vztah ani nie je, je tam vztah iba na ```orders``` nema to byt nahodou iba ```Client::where('first_name', 'ILIKE', '%' . $string . '%')``` ?

Reply:
odporucam este to co mas v metode ```json``` ako parameter vytiahnut do premennej a len tu premennu posalat ako parameter